lib/logstash/outputs/logstash.rb

```diff
@@ -54,6 +55,25 @@ class LogStash::Outputs::Logstash < LogStash::Outputs::Base
   # SSL:TUNING
   config :ssl_supported_protocols, :validate => :string, :list => true
 
+  DEFAULT_PORT = 9800.freeze
+
+  HTTP_METHOD = "post".freeze
+
+  RETRIABLE_CODES = [429, 500, 502, 503, 504]
```

Comment:
review note: these retry factors came from http-output. However, `retry_non_idempotent` isn't intentionally activated because for the failure cases, route decides which host to send the failed event.